src/recon_buildblock/PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin.cxx

```diff
@@ -122,6 +121,22 @@ set_num_subsets(const int new_num_subsets)
   return this->num_subsets;
 }
 
+template <typename TargetT>
+void
+PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin<TargetT>::
+set_proj_matrix_bybin(const shared_ptr<ProjMatrixByBin>& arg)
+{
+    this->PM_sptr = arg;
+}
+
+template<typename TargetT>
+void
+PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin<TargetT>::
+set_proj_data_info_sptr(const shared_ptr<ProjData>& arg)
```

Comment:
Not sure are all why we need this. Maybe when only using cache and no input?

If we do need it

```suggestion
set_proj_data_info(const ProjData& arg)
```
I'm not sure if we have set_proj_data_info* elsewhere. If so, follow us naming convention, else remove _sptr as it's an internal detail how it stores it.

Best to pass a red to ProjData as passing a shared pointer implies to the user we're going to do something with the pointer

Reply:
ok. I understand. It's quite ugly, but fine.

I suppose the better way would be to create our own listmode fileformat that has the projdatainfo/examinfo and your "cached" file. But that's far too much work to do now.

Reply:
Yeah, that could be a good final stage for the cache file.